src/main/java/pl/pjwstk/woloappapi/security/UserPrincipal.java

```diff
@@ -0,0 +1,89 @@
+package pl.pjwstk.woloappapi.security;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import pl.pjwstk.woloappapi.model.UserEntity;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+@Getter
+@Setter
+public class UserPrincipal implements OAuth2User, UserDetails {
+    private Long id;
+    private String email;
+    private String password;
+    private Collection<? extends GrantedAuthority> authorities;
+    private Map<String, Object> attributes;
+
+    public UserPrincipal(Long id, String email, String password, Collection<? extends GrantedAuthority> authorities) {
+        this.id = id;
+        this.email = email;
+        this.password = password;
+        this.authorities = authorities;
+    }
+
+    public static List<GrantedAuthority> convertStringToAuthorities(String role) {
+        List<GrantedAuthority> authorities = new ArrayList<>();
+
+        switch (role) {
+            case "User" -> authorities.add(UserRole.USER);
```

Comment:
nie działało:
```
case UserRole.USER -> ...
```
?

Jesli nie, to lepiej jest użyć `if` niż `case` z magic stringami.

Reply:
https://softwareengineering.stackexchange.com/questions/365339/what-is-wrong-with-magic-strings